galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/common/RasQueryParameters.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright contributors to the Galasa project 
+ */
+package dev.galasa.framework.api.ras.internal.common;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.time.*;
+
+import javax.servlet.http.HttpServletResponse;
+import javax.validation.constraints.NotNull;
+
+import dev.galasa.framework.TestRunLifecycleStatus;
+import dev.galasa.framework.api.ras.internal.verycommon.InternalServletException;
+import dev.galasa.framework.api.ras.internal.verycommon.QueryParameters;
+import dev.galasa.framework.api.ras.internal.verycommon.ServletError;
+import static dev.galasa.framework.api.ras.internal.verycommon.ServletErrorMessage.*;
+
+
+public class RasQueryParameters {
+
+	public static final int DEFAULT_PAGE_NUMBER = 1;
+	public static final int DEFAULT_NUMBER_RECORDS_PER_PAGE = 100;
+    private final Map<String, Boolean> sortDirectionMap = new HashMap<String,Boolean>(){{
+        put("asc",true);
+        put("ascending",true);
+        put("desc",false);
+        put("descending",false);
+    }};
+
+    private QueryParameters generalQueryParams ;
+
+    public RasQueryParameters(QueryParameters generalQueryParams) {
+        this.generalQueryParams = generalQueryParams;
+    }
+
+
+    // make func to validate status values
+    public List<TestRunLifecycleStatus> getStatusesFromParameters () throws InternalServletException{
+		// status values received from the query
+		List<String> queryStatuses = generalQueryParams.getMultipleString("status", null);
+		
+		if (queryStatuses != null){
+			List<TestRunLifecycleStatus> returnStatuses = new ArrayList<TestRunLifecycleStatus>();
+			for (String status : queryStatuses){
+				String statusUppercase = status.toUpperCase();
+				if (TestRunLifecycleStatus.isStatusValid(statusUppercase)) {
+					returnStatuses.add(TestRunLifecycleStatus.valueOf(statusUppercase));
+				} else {
+					ServletError error = new ServletError(GAL5014_STATUS_NAME_NOT_RECOGNIZED, status, TestRunLifecycleStatus.getAll().toString());
+					throw new InternalServletException(error, HttpServletResponse.SC_BAD_REQUEST);
+				}
+			}
+			return returnStatuses;
+		}
+
+		return null;
```

Comment:
Might be better to return an empty list ? Then far less likely to NullPointerException. And you can check list.isEmpty() ...